paddlenlp/transformers/model_utils.py

```diff
@@ -695,6 +703,54 @@ def save_pretrained(self, save_dir: str):
         else:
             logger.warning("Save pretrained model only supported dygraph mode for now!")
 
+    def save_to_hf_hub(
+        self,
+        repo_id: str,
+        private: Optional[bool] = None,
+        commit_message: Optional[bool] = None,
+        revision: Optional[str] = None,
+        create_pr: bool = False,
+    ):
+        """
+        Uploads all elements of this model to a new HuggingFace Hub repository.
+        Args:
+            repo_id (str): Repository name for your model/tokenizer in the Hub.
+            private (bool, optional): Whether the model/tokenizer is set to private
+            commit_message (str, optional) — The summary / title / first line of the generated commit. Defaults to: f"Upload {path_in_repo} with huggingface_hub"
+            revision (str, optional) — The git revision to commit from. Defaults to the head of the "main" branch.
+            create_pr (boolean, optional) — Whether or not to create a Pull Request with that commit. Defaults to False.
+                If revision is not set, PR is opened against the "main" branch. If revision is set and is a branch, PR is opened against this branch.
+                If revision is set and is not a branch name (example: a commit oid), an RevisionNotFoundError is returned by the server.
+
+        Returns: The url of the commit of your model in the given repository.
+        """
+        token = HfFolder.get_token()
+        if token is None:
+            raise ValueError(
+                "You must login to the Hugging Face hub on this computer by typing `huggingface-cli login`."
+            )
+
+        repo_url = create_repo(repo_id, token=token, private=private, exist_ok=True)
```

Comment:
There is no need to check for the token manually anymore.
If the user is not logged in and try to use `create_repo` it would raise a `ValueError`.
```
ValueError: Token is required (write-access action) but no token found. You need to provide a token or be logged in to Hugging Face with `huggingface-cli login` or `huggingface_hub.login`. See https://huggingface.co/settings/tokens.
```

The only need for an explicit `token=token` is if you set `token` as an argument of `save_to_hf_hub`, which is not the case here. So you can simplify the logic to:

```suggestion
        repo_url = create_repo(repo_id, private=private, exist_ok=True)
```

Reply:
Thanks for the suggestion